addons/l10n_fr_fec/wizard/account_fr_fec.py

```diff
@@ -153,14 +153,19 @@ def generate_fec(self):
             unaffected_earnings_results = self._do_query_unaffected_earnings()
             unaffected_earnings_line = False
 
-        sql_query = '''
+        if self.pool['account.account'].name.translate:
+            lang = self.env.user.lang or get_lang(self.env).code
+            aa_name = f"COALESCE(aa.name->>'{lang}', aa.name->>'en_US')"
```

Comment:
`get_lang` can come from the context, so this is unsafe. The code needs to be modified to follow the (query, params) pattern.

Also `COALESCE(col->>lang, col->>'en_US')` is a recurring pattern throughout the PR (for good reason), maybe we could have a utility pl/sql helper function for that e.g. `translated(column, lang)` or `_(column, lang)` or something?

Reply:
but `get_lang` compares the langues on the `get_installed()` list

Reply:
Good point. Also it then goes and get the value in `res.lang` anyway. Sorry about that.